staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -347,60 +439,64 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 	})
 	if err != nil {
 		utilruntime.HandleError(err)
-		return
+		return err
 	}
+	klog.Infof("Initializing node %s with cloud provider", nodeName)
 
-	curNode, err := cnc.kubeClient.CoreV1().Nodes().Get(context.TODO(), node.Name, metav1.GetOptions{})
+	curNode, err := cnc.nodeInformer.Lister().Get(nodeName)
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to get node %s: %v", node.Name, err))
-		return
+		utilruntime.HandleError(fmt.Errorf("failed to get node %s: %v", nodeName, err))
+		return err
 	}
 
 	cloudTaint := getCloudTaint(curNode.Spec.Taints)
 	if cloudTaint == nil {
 		// Node object received from event had the cloud taint but was outdated,
 		// the node has actually already been initialized.
-		return
+		return fmt.Errorf("this node %s is registered without the cloud taint, will not process", nodeName)
```

Comment:
This should return a nil error as well, since a node without the cloud taint means it has been initialized already, so this sync event can be ignored.